vimiv/api/settings.py

```diff
@@ -315,6 +317,80 @@ def __str__(self) -> str:
         return "String"
 
 
+class OrderSetting(Setting):
+    """Stores an ordering setting."""
+
+    typ = str
+
+    ORDER_TYPES: Dict[str, Tuple[Callable[..., Any], bool]] = {}
+
+    def convert(self, value: str) -> str:
+        if value not in self.ORDER_TYPES:
+            raise ValueError(f"Option must be one of {', '.join(self.ORDER_TYPES)}")
+        return value
+
+    def get_para(self) -> Tuple[Callable[..., Any], bool]:
+        """Returns the ordering parameters according to the current set value."""
+        try:
+            return self.ORDER_TYPES[self.value]
+        except KeyError:
+            raise ValueError(f"Option must be one of {', '.join(self.ORDER_TYPES)}")
+
+    def suggestions(self) -> List[str]:
+        return [str(value) for value in self.ORDER_TYPES]
+
+    @staticmethod
+    def _natural_sort(text: str) -> List[Union[str, int]]:
+        """Key function for natural sort.
+
+        Credits to https://stackoverflow.com/a/5967539/5464989
+        """
+
+        def convert(t: str) -> Union[str, int]:
+            return int(t) if t.isdigit() else t
+
+        return [convert(c) for c in re.split(r"(\d+)", text)]
+
+    def __str__(self) -> str:
+        return "Order"
+
+
+class ImageOrderSetting(OrderSetting):
```

Comment:
I have merged the classes in 78ad67e7879ea4250b85e8b3e62b4c401b39404a

However, I do not manage to reference the static method `_natural_sort` from the class variable `ORDER_TYPES`. Can you hint me to the right syntax?

Reply:
Ah, good point. In the current way you will not be able to access the `_natural_sort` method as it is defined later than the dictionary. I see two options:
- Just define this as a regular function, could even go into `utils` as far as I am concerned.
- Instead of defining the dictionary as a class attribute, you could have a function that returns this dictionary, something along:
```python
@classmethod
def _default_order_types(cls):
    return {
        ...
        "natural": cls._natural_sort
    }
```

Reply:
Thanks for the info, I have moved it to `utils`.

Reply:
`__init__` is great, in principle one could also sort non-file objects with this :blush: